worker/caasunitprovisioner/worker_test.go

```diff
@@ -1018,9 +1018,12 @@ func (s *WorkerSuite) TestV2CharmExitsApplicationWorker(c *gc.C) {
 	s.sendApplicationChanges(c, "gitlab")
 	waitCharmGetterCalls("ApplicationCharmInfo")
 
-	// Wait till Life() is called to synchronize (ensure charmInfo is done).
+	// Wait till application worker has started up
+	var appWorker worker.Worker
 	for a := coretesting.LongAttempt.Start(); a.Next(); {
-		if len(s.lifeGetter.Calls()) > 0 {
+		aw, ok := caasunitprovisioner.AppWorker(w, "gitlab")
```

Comment:
We need to assert that appWorker is set after we exit the loop

Reply:
Ah, thanks - fixed.